providers/packetframe/packetframeProvider.go

```diff
@@ -0,0 +1,237 @@
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+// packetframeProvider is the handle for this provider.
+type packetframeProvider struct {
+	client      *http.Client
+	baseURL     *url.URL
+	token       string
+	domainIndex map[string]zone
+}
+
+var defaultNameServerNames = []string{
+	"ns1v4.packetframe.com",
+	"ns2v4.packetframe.com",
+}
+
+// newPacketframe creates the provider.
+func newPacketframe(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	if m["apikey"] == "" {
+		return nil, fmt.Errorf("missing Packetframe token")
+	}
+
+	baseURL, err := url.Parse(defaultBaseURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid base URL for Packetframe")
+	}
+	client := http.Client{}
+
+	api := &packetframeProvider{client: &client, baseURL: baseURL, token: m["apikey"]}
+
+	return api, nil
+}
+
+var features = providers.DocumentationNotes{
+	providers.DocDualHost:            providers.Cannot(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanGetZones:            providers.Unimplemented(),
+}
+
+func init() {
+	fns := providers.DspFuncs{
+		Initializer:   newPacketframe,
+		RecordAuditor: AuditRecords,
+	}
+	providers.RegisterDomainServiceProviderType("PACKETFRAME", fns, features)
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (api *packetframeProvider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	return models.ToNameservers(defaultNameServerNames)
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (api *packetframeProvider) GetZoneRecords(domain string) (models.Records, error) {
+
+	if api.domainIndex == nil {
+		if err := api.fetchDomainList(); err != nil {
+			return nil, err
+		}
+	}
+	zone, ok := api.domainIndex[domain+"."]
+	if !ok {
+		return nil, fmt.Errorf("'%s' not a zone in Packetframe account", domain)
+	}
+
+	records, err := api.getRecords(zone.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not load records for '%s'", domain)
+	}
+
+	existingRecords := make([]*models.RecordConfig, len(records))
+
+	dc := models.DomainConfig{
+		Name: domain,
+	}
+
+	for i := range records {
+		existingRecords[i] = toRc(&dc, &records[i])
+	}
+
+	return existingRecords, nil
+}
+
+// GetDomainCorrections returns the corrections for a domain.
+func (api *packetframeProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc, err := dc.Copy()
+	if err != nil {
+		return nil, err
+	}
+
+	dc.Punycode()
+
+	if api.domainIndex == nil {
+		if err := api.fetchDomainList(); err != nil {
+			return nil, err
+		}
+	}
+	zone, ok := api.domainIndex[dc.Name+"."]
+	if !ok {
+		return nil, fmt.Errorf("'%s' not a zone in Packetframe account", dc.Name)
```

Comment:
Suggestion:

    		return nil, fmt.Errorf("no such zone %q in Packetframe account", dc.Name)

(This is just a style thing for me... I prefer error messages don't start with variable text.)